### PR TITLE
CASSANDRA-18736 - Streaming exception race creates corrupt transaction log files that p…

### PR DESCRIPTION
**Pull request by jonmeredith**
See [CASSANDRA-18736](https://issues.apache.org/jira/browse/CASSANDRA-18736)

### CODE REVIEW

File: test/unit/org/apache/cassandra/db/lifecycle/LogTransactionTest.java [hunk 1]
```diff
@@ -1359,23 +1367,62 @@ static Set<File> getFinalFiles(File folder)
...
+        }
+        catch (IOException io)
+        {
+            if (Throwables.isCausedBy(io, t -> t instanceof NoSuchFileException))
+            {
+                return Stream.empty();
+            }
+            else
+            {
+                throw new RuntimeException(io);
```
Comment (maedhroz) [inline test/unit/org/apache/cassandra/db/lifecycle/LogTransactionTest.java]:
```suggestion
                throw new UncheckedIOException(io);
```

**Review by maedhroz — APPROVED** [review]
I looked a little bit at the thread-safety of `LogFile#complete()` and `LogFile#verify()`, but it looks like the codepaths that don't lock when accessing them are in offline tools or startup.

File: test/unit/org/apache/cassandra/Util.java [hunk 2]
```diff
@@ -850,4 +852,27 @@ public static void setUpgradeFromVersion(String version)
...
     }
+
+    /**
+     * Sets the length of the file to given size. File will be created if not exist.
+     *
+     * @param file file for which length needs to be set
+     * @param size new szie
+     * @throws IOException on any I/O error.
+     */
+    public static void setFileLength(File file, long size) throws IOException
```
Comment (maedhroz) [inline test/unit/org/apache/cassandra/Util.java]: nit: unused?

Comment (jonmeredith) [reply test/unit/org/apache/cassandra/Util.java]: thx, picked it up from the refactored 4.1 version. Removing.

**Comment by jonmeredith** [comment]
Merged.

**Pull request closed without merging**